File: lecture10-morning/36_bubble_sort.py
import time

def bubble_sort(arr):
    for n in range(len(arr) - 1, 0, -1):
        for i in range(n):
            if arr[i] > arr[i + 1]:
                arr[i], arr[i + 1] = arr[i + 1], arr[i]
                
# Sample list to be sorted
arr = [39, 12, 18, 85, 72, 10, 2, 18]
print("Unsorted list is:")
print(arr)

start = time.time()
bubble_sort(arr)
elapsed = time.time() - start

print("Sorting time in milliseconds is: " + str(elapsed * 1000))

File: lecture11-morning/code/c/36_bubble_sort.c
#include <stdio.h>
#include <time.h>


void swap(int* arr, int i, int j) {
  // TODO: complete this.
}

void bubbleSort(int arr[], int n) {
    for (int i = 0; i < n - 1; i++) {
        for (int j = 0; j < n - i - 1; j++) {
          // TODO: complete this.
    }
}

int main() {
  int arr[] = {39, 12, 18, 85, 72, 10, 2, 18};
  int n = sizeof(arr) / sizeof(arr[0]);

  clock_t start, end;
  double cpu_time_used;

  start = clock();
  bubbleSort(arr, n);
  double elapsed = ((double) (clock() - start)) * 1000 / CLOCKS_PER_SEC;
  printf("Elapsed time: %.20f", elapsed);
  
  return 0;
}
